components/tidb_query/src/rpn_expr/impl_arithmetic.rs

```diff
@@ -466,30 +461,17 @@ fn int_divide_decimal(
     lhs: &Option<Decimal>,
     rhs: &Option<Decimal>,
 ) -> Result<Option<Int>> {
-    use crate::codec::mysql::Res;
-
-    if lhs.is_none() || rhs.is_none() {
-        return Ok(None);
-    }
-    let lhs = lhs.as_ref().unwrap();
-    let rhs = rhs.as_ref().unwrap();
-
-    match lhs / rhs {
-        Some(v) => match v {
-            Res::Ok(v) => match v.as_i64() {
-                Res::Ok(v_i64) => Ok(Some(v_i64)),
-                Res::Truncated(v_i64) => Ok(Some(v_i64)),
-                Res::Overflow(_) => {
-                    Err(Error::overflow("BIGINT", &format!("({} / {})", lhs, rhs)).into())
-                }
-            },
-            Res::Truncated(_) => Err(Error::truncated().into()),
-            Res::Overflow(_) => {
-                Err(Error::overflow("DECIMAL", &format!("({} / {})", lhs, rhs)).into())
-            }
-        },
-        None => Ok(ctx.handle_division_by_zero().map(|()| None)?),
-    }
+    let result = try_opt!(arithmetic_with_ctx::<DecimalDivide>(ctx, lhs, rhs));
+    Ok(result
+        .as_i64()
+        .into_result_with_overflow_err(
```

Comment:
does as_i64 failed need be handle truncated
 `Res::Truncated(v_i64) => Ok(Some(v_i64)),`